src/ctia/entity/incident.clj

```diff
@@ -159,7 +160,50 @@
            :intended_effect
            :assignees
            :promotion_method
-           :severity]))
+           :severity
+           :tactics]))
+
+(comment
+  (defn generate-mitre-tactic-scores
+    "script for stripping proprietary info from mitre tactic scores.
+    Use to generate new :remappings for :tactics sorting."
+    [csv-file]
+    (let [s (slurp csv-file)
+          ;; lifecycle order
+          relevant-tactics ["TA0043"
+                            "TA0042"
+                            "TA0001"
+                            "TA0002"
+                            "TA0003"
+                            "TA0004"
+                            "TA0005"
+                            "TA0006"
+                            "TA0007"
+                            "TA0008"
+                            "TA0009"
+                            "TA0011"
+                            "TA0010"
+                            "TA0040"]
+          tactic->pos (into {} (map-indexed (fn [i id] [id i]))
+                            relevant-tactics)
+          groups (-> s 
+                     ((requiring-resolve 'cheshire.core/parse-string))
+                     ((requiring-resolve 'clojure.walk/keywordize-keys))
+                     (->> (filter (comp (set relevant-tactics) :id))
+                          (group-by :risk_score)
+                          (sort-by key)
+                          (map second)))
+          out (into (sorted-map-by (fn [id1 id2]
+                                     (< (tactic->pos id1 0) (tactic->pos id2 0))))
+                    (map (fn [score group]
+                           (zipmap (map :id group) (repeat score)))
+                         (next (range)) groups))
+
+          _ (assert (= relevant-tactics (keys out))
```

Comment:
I know this is totally not production code, but it's the last binding in the block man, how can that not hurt your eyes?

Reply:
I like making side effects stand out :) It hurts my brain less.